R/logging.R

```diff
@@ -0,0 +1,110 @@
+#' Get node log files from Azure Storage. By default, this operation will print the files on screen.
+#'
+#' @param cluster The cluster object
+#' @param nodeId Id of the node
+#' @param type The type of logs the user wants: "stdout" or "stderr"
+#' @param verbose Flag for printing log files onto console
+#'
+#' @param ... Further named parameters
+#' \itemize{
+#'  \item{"localPath"}: { Path to save file to }
+#'  \item{"overwrite"}: { Will only overwrite existing localPath }
+#'}
+#'
+#' @examples
+#' stdoutText <- getClusterLogs(cluster, "tvm-1170471534_1-20170829t072146z", type = "stdout", verbose = FALSE)
+#' getClusterLogs(cluster, "tvm-1170471534_2-20170829t072146z", type = "stderr", localPath = "abc.txt", overwrite = TRUE)
+#' @export
+getClusterLogs <-
+  function(cluster,
+           nodeId,
+           type = c("stdout", "stderr"),
+           verbose = TRUE,
+           overwrite = FALSE,
+           localPath = NULL) {
```

Comment:
**soft opinion** :

should automatically save to disk for them? i feel like the more normal behavior is to just return the logs in the r-session so that they can do what they want with it (print it, write to disk, etc)

i'm not sure on this one.

Reply:
Yeah, this is a tricky one. I'm not sure how the API in R works, but in theory this can be a pretty big log/file. If that is the case then it may make more sense to write the file to disk (ideally in chunks) so it doesn't take up a ton of resources (like RAM). Not sure if this is too contrived though since I have no clue how large a log file gets on average.